doc/perf.md

```diff
@@ -190,16 +183,15 @@ It is often easier to work with optimizations disabled (or with `-O1`),
 for two reasons:
 1. Optimizing a large program is slow. The optimized program will run faster,
     but it takes a while for LLVM to figure out how to make the code faster.
-    Waiting for the optimizer to do its thing is annoying.
-2. The optimizer assumes that your code doesn't do some dumb things.
-    This can have surprising consequences.
+    Waiting for the optimizer to do its thing is annoying. See Example #1
+2. The optimizer assumes that your code doesn't contain dumb things, that cause **undefined behavior** (see below). If it does it may behave unpredictably (not crashing when it should do so, etc.). See Example #2
```

Comment:
```suggestion
2. The optimizer assumes that your code doesn't contain dumb things, or specifically, things that cause **undefined behavior** (see below). If it does, it may behave unpredictably (crashing, some parts of the code not running, etc.). See Example #2 below.
```

Reply:
I think you should keep the part with 'not crashing when it should do'. That's the unusal thingy and the consequence of the optimizations. Crashing and not running is something the program would do without it as well.

Reply:
```suggestion
2. The optimizer assumes that your code doesn't contain dumb things, or specifically, things that cause **undefined behavior** (see below). If it does, it may behave unpredictably (crashing, not crashing as expected, some parts of the code not running, etc.). See Example #2 below.
```

Reply:
How about 'If it does, it may behave unpredictably (crashing, not crashing as expected, some parts of the code not running, etc.). See Example #2 below.'

Reply:
I think that's exactly what I suggested.